src/components/Combobox/Combobox.jsx

```diff
@@ -93,14 +163,48 @@ const Combobox = forwardRef(
     );
 
     const filterdOptions = useMemo(() => {
-      return options.filter(({ label }) => !filterValue || label.includes(filterValue));
-    }, [options, filterValue]);
+      return options.filter(({ label }) => !filterValue || label.toLowerCase().includes(filterValue.toLowerCase()));
+    }, [options, categories, filterValue]);
 
     const renderedItems = useMemo(() => {
-      return filterdOptions.map((option, index) => {
-        return renderOption(index, option, activeItemIndex === index, isActiveByKeyboard, onOptionClick, onOptionHover);
-      });
-    }, [filterdOptions, activeItemIndex, isActiveByKeyboard, onOptionClick, onOptionHover]);
+      if (categories) {
+        const optionsByCategories = getOptionsByCategories(filterdOptions, categories, filterValue);
+
+        let index = 0;
```

Comment:
why index? the Array.map function has index as the seconday parameter https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
thats what I did at first! but then it didn't work because I need to keep the same index on all options (from different categories).
when I used the map index I got indexes like 1,1,2,3,1,2 and the keyboard navigation didn't work